### PR TITLE
[HEI-342] Dev/architecture v11 default role

### PR DESCRIPTION
**Pull request by TonyKatkov89**

### CODE REVIEW

File: hypex/dataset/abstract.py [hunk 1]
```diff
@@ -117,6 +120,32 @@ def search_columns(
             )
         ]
 
+    def replace_roles(
+        self,
+        to_replace_roles: Union[ABCRole, str, List[Union[ABCRole, str]]],
```
Comment (Dmatryus) [inline hypex/dataset/abstract.py:125]: Dict

File: hypex/dataset/abstract.py [hunk 2]
```diff
@@ -171,3 +208,29 @@ def get_values(
         column: Optional[str] = None,
     ) -> Any:
         return self._backend.get_values(row=row, column=column)
+
+    def set_roles(
```
Comment (Dmatryus) [inline hypex/dataset/abstract.py:212]: Universal mechanism for init and set, maybe replace.

**Review by Dmatryus — CHANGES_REQUESTED** [review]
Universalisation is power! 🦾

**Review by VasilievaKA — APPROVED** [review]
Perfect!

**Review by Dmatryus — APPROVED** [review]
👍

**Pull request merged**